packages/react/src/publication/adapters/ReportPublicationPresenter.ts

```diff
@@ -0,0 +1,17 @@
+import {
+  AlreadyReportedError,
+  IReportPublicationPresenter,
+} from '@lens-protocol/domain/use-cases/publications';
+import { Deferred, Result } from '@lens-protocol/shared-kernel';
+
+export class ReportPublicationPresenter implements IReportPublicationPresenter {
```

Comment:
Could we reuse our `PromiseResultPresenter` presenter and avoid having `ReportPublicationPresenter` at all?